avocado/utils/astring.py

```diff
@@ -219,7 +219,7 @@ def string_safe_encode(string):
     try:
         return string.encode("utf-8")
     except UnicodeDecodeError:
-        return string.decode("utf-8").encode("utf-8")
+        return string.decode("utf-8", "ignore").encode("utf-8")
```

Comment:
Actually the `UnicodeDecodeError` is the proper one as it fails to decode the string in order to encode it... Anyway, sending v2 with improved selftest.